pkg/util/deployer/deployer.go

```diff
@@ -103,19 +103,41 @@ func (depl *deployer) Remove(ctx context.Context, data interface{}) error {
 	}
 
 	var errs []error
+	namespaceName := ""
 	for _, obj := range resources {
-		// delete any deployments we have
-		if deployment, ok := obj.(*appsv1.Deployment); ok {
-			err := depl.dh.EnsureDeleted(ctx, "Deployment", deployment.Namespace, deployment.Name)
-			// Don't error out because then we might delete some resources and not others
-			if err != nil {
-				errs = append(errs, err)
-			}
+		// remove everything we created that has name and ns
+		nameValue, err := getField(obj, "Name")
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		nsValue, err := getField(obj, "Namespace")
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		name := nameValue.String()
+		ns := nsValue.String()
+		if reflect.TypeOf(obj).String() == "*v1.Namespace" {
+			// dont delete the namespace for now
+			namespaceName = name
+			continue
+		}
+		errDelete := depl.dh.EnsureDeletedGVR(ctx, obj.GetObjectKind().GroupVersionKind().GroupKind().String(), ns, name, "")
+		if errDelete != nil {
+			errs = append(errs, errDelete)
+		}
+	}
+	if namespaceName != "" {
+		// remove the namespace
+		err := depl.dh.EnsureDeleted(ctx, "Namespace", "", namespaceName)
+		if err != nil {
+			errs = append(errs, err)
```

Comment:
could we have this in a function? would avoid having all these continue

Reply:
sure